Update SIG info (contact info / charter)

justaugustus opened the pull request:
A wild SIG charter has appeared...

/assign @calebamiles @jdumars @tpepper
Signed-off-by: Stephen Augustus <foo@agst.us>

jdumars commented on charter.md:
```diff
@@ -0,0 +1,104 @@
+
+Additionally, SIG Release has a concept of membership. SIG members can be occasionally called on to assist with decision making, especially as it relates to gathering historical context around existing policies.
+
+While a strict definition of membership has yet to be established, we can initially seed a set of members from the list of Release Team members from Kubernetes 1.10 through to Kubernetes 1.13.
```
IMO this should be all documented former release team members.

justaugustus replied:
Yep. That sounds good. I think my hesitancy around saying that was having to build the list.
How can we ensure that that list remains fresh i.e., active SIG Release members as opposed to the entire historical set? That also raises the question of how we classify an active member. 

I was also thinking of adding a `sig-release-members` entry in `OWNERS_ALIASES`. Thoughts?

dims replied:
@justaugustus since OWNERS_ALIASES is for review/approvals, we should may be just use a github group. would be easy to add/drop people too.

dims replied:
+1 to `all documented former release team members`

jdumars commented on charter.md:
```diff
@@ -0,0 +1,104 @@
+- Status: Implemented
+- Membership: Rotating
+
+#### Product Security Team (Subproject)
```
Is this really a subproject under release?

justaugustus replied:
:thinking:

justaugustus replied:
I thought it was, but allow me to get some clarity there.

jdumars commented on charter.md:
```diff
@@ -0,0 +1,104 @@
+  - Define and collect metrics related to the release in order to measure progress over each release
+- Collaboration with downstream communities which build artifacts from Kubernetes releases
+  - Ensure the appropriate level of integration with publishing the OSS build artifacts
+
```
Conduct release retrospectives

justaugustus replied:
Hmmm, I'm thinking this [sh|c]ould be a function of SIG PM. I'll add language that it should be facilitated by both.

justaugustus replied:
Added.

justaugustus commented:
Announced on `k-release`, `k-release-leads`, and `steering`: https://groups.google.com/d/topic/kubernetes-sig-release/CIe4ekG7GkE/discussion
Maintaining lazy consensus hold until Friday, November 2nd, EOD PT. 
/hold

justaugustus commented:
Adding our Steering liaisons / reviewers.
/assign @timothysc @spiffxp 
/committee steering

rel: https://github.com/kubernetes/steering/issues/31

dims commented on charter.md:
```diff
@@ -0,0 +1,105 @@
+  - Define and collect metrics related to the release in order to measure progress over each release
+  - Facilitate release retrospectives in conjunction with SIG PM
+- Collaboration with downstream communities which build artifacts from Kubernetes releases
+  - Ensure the appropriate level of integration with publishing the OSS build artifacts
```
i'd drop `OSS` here. it's a bit jarring. by definition the work being done in this sig is Open Source.

justaugustus replied:
Will do.

dims commented on charter.md:
```diff
@@ -0,0 +1,105 @@
+## Scope
+
+- Production of high quality Kubernetes releases on a reliable schedule
+- Ensure there is a consistent group of community members in place to support the release process
```
how do we add `across time` concept here? one of the problems we have right now is that all sig meetings are mostly task based for the specific release and we hardly spend time on things that need to be done across releases.

justaugustus replied:
IMO, current RT topics should be contained to RT / Burndown meetings, except where it is required to clarify existing policy or help define new policy.

timothysc commented on charter.md:
```diff
@@ -0,0 +1,105 @@
+
+## Scope
+
+- Production of high quality Kubernetes releases on a reliable schedule
```
What does high quality mean, that's very subjective.

timothysc commented on charter.md:
```diff
@@ -0,0 +1,105 @@
+
+As of the initial draft of this charter, the following are the proposed subprojects / WGs for SIG Release:
+
+#### Release Team (Subproject)
```
Subprojects are not typically called out in the charter, b/c they are subject to change.  The charter spells out the mission and details of the subproject(s) are simply added to the community repo and align with that mission.

jberkus commented on charter.md:
```diff
@@ -0,0 +1,105 @@
+    - Binary distributions
+    - Release notes
+- Continually improving release and development processes
+  - Work closely with SIG Contributor Experience to define and build tools to facilitate release process (e.g. dashboards)
```
add:
- work closely with SIG Testing to determine and implement tests, automation, and labeling required for stable releases

jberkus commented on README.md:
```diff
@@ -1,38 +1,19 @@
-## Join us!
-- [Google group](https://groups.google.com/forum/#!forum/kubernetes-sig-release)
-- [Slack channel](https://kubernetes.slack.com/messages/C2C40FMNF/)
-- [Events and meetings calendar](https://calendar.google.com/calendar/embed?src=coreos.com_regcvcrgvq98lua2ikijg1g1uk%40group.calendar.google.com&ctz=America/Los_Angeles)
```
Hey, this deletes all meeting and contact information, and that meeting information is NOT replicated in the charter.  IMHO, we should leave meeting and contact information in the README.

justaugustus commented on README.md:
```diff
@@ -1,38 +1,19 @@
-  - Define and collect metrics related to the release in order to measure progress over each release
-- Collaboration with downstream communities which build artifacts from Kubernetes releases
-  - Ensure the appropriate level of integration with publishing the OSS build artifacts
+## Join us!
```
> Hey, this deletes all meeting and contact information, and that meeting information is NOT replicated in the charter. IMHO, we should leave meeting and contact information in the README.

@jberkus -- the contact info is here. It just got pushed a little lower in the diff.

justaugustus commented:
I think I've addressed everyone's comments. Most of it was just tweaking language.

> There are some things that should be removed, but I'd also like to see an addition of what where things live that you currently own. e.g. the entire instructions for the release process

@timothysc -- I've removed the subprojects section. `k/sig-release` is the canonical location for all things SIG Release related, so instructions on the relevant processes can be found here. Processes related to the Release Team, for example, would be found in [k/sig-release/release-team](https://git.k8s.io/sig-release/release-team).

I would want additional SIG Release subprojects / WGs to follow the same structure:
```
k/sig-release
   |- release-team
   |- release-engineering
   |- licensing
   |- lts  
```

If you're specifically referring to build / release processes, instead of putting that information in the charter, I'd want to formally turn up the Release Engineering team and then task them with building out the documentation (as I know precious little about that process and it's also something that would be subject to change).

Does that sound reasonable to you?

timothysc commented on charter.md:
```diff
@@ -0,0 +1,74 @@
+    (e.g. burndown meetings, cutting beta releases) with the intent of meeting the release schedule
+  - Managing the creation of release specific artifacts, including:
+    - Code branches
+    - Binary distributions
```
s/distributions/artifacts

justaugustus replied:
Fixed!

justaugustus commented:
Thanks for the review, @timothysc!

@spiffxp -- this is ready for a final pass :)
cc: @calebamiles @tpepper

justaugustus commented:
Removing lazy consensus hold.
/hold cancel

justaugustus commented:
Thanks everyone who reviewed this!
Ready for a final `/approve`.

timothysc commented:
@spiffxp for final comments.

calebamiles commented on charter.md:
```diff
@@ -0,0 +1,74 @@
+
+### Subproject Creation
+
+- Subprojects must be created by [KEP] proposal and accepted by [lazy-consensus] with fallback on majority vote of
```
I believe it that subproject creation, along with most other major decisions, should follow the same process:

1. seek lazy consensus around a KEP
1. in the event consensus cannot be reached -> majority decision by SIG Chairs
1. SIG members _may_ override the majority decision of SIG Chairs by a supermajority (60%)

this may sound like process for process sake but I really feel that it's important to take the position

*  the people who do the work make the decisions (membership is derived from `OWNERS` and the ultimate authority is held my SIG members)
* SIG Chairs are supposed to represent the sentiment of and facilitate the decision making by SIG Members

justaugustus replied:
@calebamiles -- Added that language.

justaugustus replied:
@tpepper -- the way I imagine the RT is a single subproject with rotating membership and the current RT Lead + Shadow acting as subproject chairs. SIG Release Chairs would be considered subproject chairs here for the sake of continuity across release cycles.

Let's bounce ideas around for that though when we draft the KEP for RT.

calebamiles commented on charter.md:
```diff
@@ -0,0 +1,74 @@
+  - Working with downstream communities responsible for packaging Kubernetes releases
+  - Working with other SIGs to agree upon the responsibilities of their SIG with respect to the release
+  - Defining and collecting metrics related to the release in order to measure progress over each release
+  - Facilitating release retrospectives in conjunction with SIG PM
```
I don't think SIG PM really plays a role here. Certainly @jdumars has but I don't think there is a formalized link here between SIG PM and SIG Release so I would remove that part

jdumars replied:
I agree with Caleb here. Another SIG should not be a potential blocker for one of the most established and arguably important continuous-improvement activities in the project. In cases where I was not able to do the retro because I was the release lead, SIG ConribX staffed it, not PM.

justaugustus replied:
Thanks for the additional context, y'all. Updated.

calebamiles commented on charter.md:
```diff
@@ -0,0 +1,74 @@
+  - Defining and collecting metrics related to the release in order to measure progress over each release
+  - Facilitating release retrospectives in conjunction with SIG PM
+- Collaborating with downstream communities which build artifacts from Kubernetes releases
+  - Ensuring the appropriate level of integration with publishing the build artifacts
```
I don't think this sentence means anything so I would remove

calebamiles commented on charter.md:
```diff
@@ -0,0 +1,74 @@
+
+### Deviations from [sig-governance]
+
+- SIG Release does not have top-level SIG Technical Leads; SIG Release Chairs can serve in both an operational and technical capacity
```
I would caveat this by saying that we don't _currently_ have any technical leads in deference to subproject leads

justaugustus replied:
The intent was that we don't require Technical Leads, either now or in the future, as our Chairs are capable of executing technical duties, as well as the fact that we'll have multiple subproject chairs for specific areas. Do you feel that's captured here or disagree with that sentiment?

jdumars replied:
Asking the SIG release chair to be "technical" is not inclusionary language. It also implies vague powers for technical arbitration that rightly belongs to sub-projects. Chairs are responsible for the organization, facilitation, and administrivia of the umbrella group. In that regard, SIG Release is not different than any other SIG WRT chairs. I would strike this language.

justaugustus replied:
Yeah, realizing now that the phrasing didn't match my intent. So stricken.

calebamiles commented on charter.md:
```diff
@@ -0,0 +1,74 @@
+
+- Subprojects must be created by [KEP] proposal and accepted by [lazy-consensus] with fallback on majority vote of
+  SIG Chairs.  The result *SHOULD* be supported by the majority of SIG members.
+  - KEP *MUST* establish subproject chairs
```
I don't think the legalistic `MUST` or `SHOULD` add much value here

calebamiles commented on charter.md:
```diff
@@ -0,0 +1,74 @@
+  - KEP *MUST* establish subproject chairs
+  - [sigs.yaml] *MUST* be updated to include subproject information and OWNERS files with subproject chairs
+  - Where subprojects processes differ from the SIG governance, they must document how
+    - e.g., if subprojects release separately - they must document how release and planning is performed
```
I would remove this because I think this could simply be discussed in the context of a KEP were the need to come up

justaugustus replied:
Removed.

calebamiles commented on charter.md:
```diff
@@ -0,0 +1,74 @@
+
+While a strict definition of membership has yet to be established, we will initially seed a set of members from the list of all documented former Release Team members and subproject OWNERS.
+
+SIG Release Members should be the first group of community members considered when seeking to fill additional roles across SIG Release proper and SIG Release subprojects, except where explicitly defined in other policy.
```
I would be comfortable taking a harder line here. Unless otherwise agreed (release team shadow roles are a notable existing example of this), _all_ official SIG roles may only be filled by SIG members

tpepper replied:
Is there a universal benefit in doing so?  It feels to me that being exclusive in this regard would be the opposite of acting as an inclusive and open community.

justaugustus replied:
@calebamiles -- Happy to take a harder line here as well, and I've updated the language to reflect that.

@tpepper -- I agree with Caleb here. There are multiple ingress points to SIG membership, all of which require some active contribution. I don't feel that they're unnecessarily restrictive or exclusive.

Examples:
- Joe serves as a Release Team shadow and subsequently becomes a role lead. At the end of that cycle, he'd officially become a SIG Release member.
- Liz contributes to the Release Engineering tooling and subsequently graduates to the OWNERS file for the release-engineering subproject. Liz would also officially become a SIG Release member at that point.

Both examples show sustained commitment to SIG Release, and I believe that that slightly higher bar is paramount for contributors interested in defining policy and process for the systems, tooling, and teams that help us release Kubernetes.

tpepper replied:
OK I can accept that reasoning.

calebamiles commented on charter.md:
```diff
@@ -0,0 +1,74 @@
+
+Additionally, SIG Release has a concept of membership. SIG members can be occasionally called on to assist with decision making, especially as it relates to gathering historical context around existing policies.
+
+While a strict definition of membership has yet to be established, we will initially seed a set of members from the list of all documented former Release Team members and subproject OWNERS.
```
see previous note about taking `approvers` from all SIG Release subprojects, making sure that previous holders of a release team roles are included retroactively. I think that the current practice of adding leads to the `OWNERS_ALIASES` ensure things are ok moving forward

justaugustus replied:
Added.

justaugustus commented on charter.md:
```diff
@@ -0,0 +1,74 @@
+
+#### SIG Membership
+
+Additionally, SIG Release has a concept of membership. SIG members can be occasionally called on to assist with decision making, especially as it relates to gathering historical context around existing policies.
```
Added more explicit criteria for SIG membership.

justaugustus commented:
@calebamiles / @tpepper -- I've addressed your comments. PTAL when you have a chance.
I'm dropping a hold on this again, to ensure we have consensus with the Chairs before merging.

/hold

jdumars commented:
I'll let @calebamiles clear his change request, and then we can LGTM.

calebamiles commented:
/lgtm
